workflows/reference_files/2.4/reference_files.smk

```diff
@@ -598,3 +598,105 @@ rule install_sigprofiler_genome:
+        outfile_temp="downloads/igblast/imgt_database/human/vdj/imgt_human_{wildcards.subchain}.html" && 
+        outfile_ttemp="downloads/igblast/imgt_database/human/vdj/imgt_human_{wildcards.subchain}.txt" && 
+        outfile="downloads/igblast/imgt_database/human/vdj/imgt_human_{wildcards.subchain}.fasta" && 
+        wget -cO $outfile_temp $URL && 
+        awk '/<pre>/{{i++}}/<\/pre>/{{j++}}{{if(i==2 && j==1 && $0 !~ "^<pre>" && $0 ~ "^>"){{print $0}}}}{{if(i==2 && j==1 && $0 !~ "^<pre>" && $0 !~ "^>"){{system("echo "$0" | tr '[:lower:]' '[:upper:]'")}}}}' $outfile_temp > $outfile_ttemp && 
+        rm $outfile_temp && 
+        edit_imgt_file.pl $outfile_ttemp > $outfile && 
+        rm $outfile_ttemp 
+        """)
+
+IMGT_CHAIN_KEY = {
+    "ig_v" : ["IGHV","IGLV","IGKV"],
+    "ig_d" : ["IGHD"],
+    "ig_j" : ["IGHJ","IGLJ","IGKJ"],
+    "tr_v" : ["TRAV","TRBV","TRDV","TRGV"],
+    "tr_d" : ["TRBD","TRDD"],
+    "tr_j" : ["TRAJ","TRBJ","TRDJ","TRGJ"]
+}
+
+rule _combine_imgt_files:
+    input:
+        lambda w: expand(rules._get_imgt_database.output.vdj, subchain = IMGT_CHAIN_KEY[w.chain])
+    output:
+        unfiltered = temp("genomes/no_build/igblast/fasta/imgt/imgt_human_{chain}.unfiltered")
+    shell:
+        op.as_one_line("""
+        cat {input} > {output}
+        """)
+
+rule _remove_imgt_dups:
+    input:
+        unfiltered = rules._combine_imgt_files.output.unfiltered
+    output:
+        filtered = "genomes/no_build/igblast/fasta/imgt_human_{chain}.fasta"
+    run:
+        fasta_file = input.unfiltered
+        output_file = output.filtered
+        if os.path.isfile(fasta_file):
+            with open(fasta_file, 'r') as handle:
+                out = open(output_file, 'w')
+                genes = []
+                keep = False
+                for line in handle:
+                    line = line.rstrip("\n")
+                    if line.startswith(">"):
+                        keep = False
+                        gene = line.lstrip(">")
+                        if gene not in genes:
+                            genes.append(gene)
+                            keep = True
+                            out.write(line + "\n")
+                    elif not line.startswith(">"):
+                        if keep:
+                            out.write(line + "\n")
+                out.close()
+
+rule _create_imgt_database:
+    input:
+        fasta = rules._remove_imgt_dups.output
+    output:
+        database = expand("genomes/no_build/igblast/database/imgt_human_{{chain}}.{ext}", ext = ['ndb','nhr','nin','nog','nos','not','nsq','ntf','nto'])
+    conda: CONDA_ENVS["igblast"]
+    shell:
+        op.as_one_line("""
+        fasta_file={input.fasta} && 
+        output_file="genomes/no_build/igblast/database/imgt_human_{wildcards.chain}" && 
+        makeblastdb -parse_seqids -dbtype nucl -in $fasta_file -out $output_file
+        """)
+
+rule _imgt_db_success:
```

Comment:
There's another Snakefile that can be used to generate all reference files prior to doing any runs. 
https://github.com/LCR-BCCRC/lcr-modules/blob/0cf2b50b501806be1d1a507a7c0e1eb91323c8ae/workflows/reference_files/2.4/prepare_reference_files.smk#L35-L68

Can you please add the output of this rule to the input of that one? You can add  the possible chain values to the config following the example of the other `config["wildcard_values"]["..."]` examples there.

Reply:
Added the `rules._imgt_db_success.output.txt` rule to `prepare_reference_files.smk` rule all